src/main/resources/validationrules/civil/civil-judicial-org-role-mapping.drl

```diff
@@ -542,4 +1002,87 @@ then
       .authorisations($joh.getTicketCodes())
       .build());
       logMsg("Rule : civil_judge_org_role_for_fee_paid_judge");
+end;
+
+/*
+ * CIVIL "judge" org role can be created by any existing judicial office holder having
+ * "CIVIL Deputy Judge-Fee-Paid, CIVIL Recorder-Fee-Paid,CIVIL Deputy District Judge - Sitting in Retirement-Fee-Paid" business role.
+ * Added in CIVIL Deputy Circuit Judge-Fee-Paid from v11
+ */
+rule "v21_civil_judge_org_role_for_fee_paid_judge"
+when
+  $f:  FeatureFlag(status && flagName == FeatureFlagEnum.CIVIL_WA_2_1.getValue())
+  $joh: JudicialOfficeHolder(office in ("CIVIL Deputy District Judge-Fee-Paid", "CIVIL Recorder-Fee-Paid",
+                                        "CIVIL Deputy District Judge - Sitting in Retirement-Fee-Paid",
+                                        "CIVIL District Judge (sitting in retirement)-Fee-Paid",
+                                        "CIVIL Tribunal Judge-Fee-Paid",
+                                        "CIVIL Circuit Judge (sitting in retirement)-Fee-Paid",
+                                        "CIVIL Deputy Circuit Judge-Fee-Paid"))
+  $bk: JudicialBooking(userId == $joh.userId)
+then
+   Map<String,JsonNode> attribute = new HashMap<>();
+   attribute.put("jurisdiction", JacksonUtils.convertObjectIntoJsonNode("CIVIL"));
+   attribute.put("primaryLocation", JacksonUtils.convertObjectIntoJsonNode($bk.getLocationId() != null ?
+   $bk.getLocationId():$joh.getPrimaryLocation()));
+   attribute.put("baseLocation", JacksonUtils.convertObjectIntoJsonNode($bk.getLocationId()));
+   attribute.put("region", JacksonUtils.convertObjectIntoJsonNode($bk.getRegionId()));
+   attribute.put("contractType", JacksonUtils.convertObjectIntoJsonNode("Fee-Paid"));
+   attribute.put("workTypes", JacksonUtils.convertObjectIntoJsonNode("decision_making_work,applications"));
+
+  insert(
+      RoleAssignment.builder()
+      .actorIdType(ActorIdType.IDAM)
+      .actorId($joh.getUserId())
+      .roleCategory(RoleCategory.JUDICIAL)
+      .roleType(RoleType.ORGANISATION)
+      .roleName("judge")
+      .grantType(GrantType.STANDARD)
+      .classification(Classification.PUBLIC)
+      .readOnly(false)
+      .beginTime($bk.getBeginTime())
+      .endTime($bk.getEndTime())
+      .attributes(attribute)
+      .authorisations($joh.getTicketCodes())
+      .build());
+      logMsg("Rule : v21_civil_judge_org_role_for_fee_paid_judge");
+end;
+
+
+/*
+ * CIVIL "recorder" org role can be created by any existing judicial office holder having
+ * "CIVIL Recorder-Fee-Paid business role.
+ */
+rule "civil_recorder_org_role"
+when
+  $f:  FeatureFlag(status && flagName == FeatureFlagEnum.CIVIL_WA_1_0.getValue())
```

Comment:
for `rule "civil_recorder_org_role"` - should this be using new flag? as it's a new role added ?

```suggestion
  $f:  FeatureFlag(status && flagName == FeatureFlagEnum.CIVIL_WA_2_1.getValue())
```

Reply:
addressed in [1b799af](https://github.com/hmcts/am-org-role-mapping-service/pull/2147/commits/1b799af3c68e411a5eca17341995a224bf5fcc74)

Reply:
Good spot @abul-syed many thanks on finding that one.